portfolio/src/main/java/com/google/sps/servlets/NewMessageServlet.java

```diff
@@ -46,7 +50,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.sendRedirect("/chat.html");
   }
 
-  private Message getMessage(HttpServletRequest request) {
-    return new Message(request.getParameter("name"), request.getParameter("text"));
+  /**
+   * Returns the nickname of the user with id, or empty String if the user has not set a nickname.
+   */
+  private String getUserNickname(String id) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("UserInfo").setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    if (entity == null) {
+      return "";
+    }
+    String nickname = (String) entity.getProperty("nickname");
+    return nickname;
```

Comment:
This seems to be used in different places, maybe extract this into a static method of a class, maybe in the com.google.sps.data package.